### PR TITLE
Remove all for key

### PR DESCRIPTION
**Pull request by groovz**
Added the function removeAllForKey in order to remove all registered and unregistered formats in memory and in disk by a specific key. asked in #243

### CODE REVIEW

**Comment by dcharbonnier** [comment]
thank you for the PR, can you add test for this feature ?

File: Haneke/Cache.swift [hunk 1]
```diff
@@ -129,6 +129,51 @@ public class Cache<T: DataConvertible where T.Result == T, T : DataRepresentable
...
+            dispatch_group_enter(group)
+            diskCache.removeData(key) {
+                dispatch_group_leave(group)
+            }
+        }
+        
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)) {
+            let timeout = dispatch_time(DISPATCH_TIME_NOW, Int64(60 * NSEC_PER_SEC))
+            if dispatch_group_wait(group, timeout) != 0 {
+                Log.error("removeData timed out waiting for disk caches")
```
Comment (dcharbonnier) [inline Haneke/Cache.swift]: it should be "*removeAllForKey* timed out waiting for disk caches"

Comment (groovz) [inline Haneke/Cache.swift]: that was changed accordingly.

**Comment by groovz** [comment]
Currently I don't have time to add the tests.

**Comment by dcharbonnier** [comment]
thx, if someone has time it would be nice so we can merge, I'm also too busy right now

**Comment by dcharbonnier** [comment]
please make the build pass and reopen if possible

**Pull request closed without merging**